Explorer/Assets/Scripts/Global/Dynamic/RealmController.cs

```diff
@@ -138,14 +140,34 @@ public async UniTask RestartRealmAsync(CancellationToken ct)
         public async UniTask<bool> IsReachableAsync(URLDomain realm, CancellationToken ct) =>
             await webRequestController.IsReachableAsync(ReportCategory.REALM, realm.Append(new URLPath("/about")), ct);
 
-        public async UniTask<AssetPromise<SceneEntityDefinition, GetSceneDefinition>[]> WaitForFixedScenePromisesAsync(CancellationToken ct)
+        public async UniTask<IReadOnlyList<SceneEntityDefinition>> WaitForFixedSceneEntityDefinitionsAsync(CancellationToken ct)
         {
             FixedScenePointers fixedScenePointers = default;
 
             await UniTask.WaitUntil(() => GlobalWorld.EcsWorld.TryGet(realmEntity, out fixedScenePointers)
                                           && fixedScenePointers.AllPromisesResolved, cancellationToken: ct);
 
-            return fixedScenePointers.Promises!;
+            List<SceneEntityDefinition> returnList = new List<SceneEntityDefinition>();
```

Comment:
Let's provide an initial capacity for the list at least to avoid re-allocation on adding

Reply:
Tackled at https://github.com/decentraland/unity-explorer/pull/2229/commits/e381e282b9d5162543a8dc749783f61aa85252a5